File: app/models/attachment.go
package models

type Attachment struct {
	ID              uint64 `gorm:"column:id;primaryKey" json:"id"`
	StorageMode     string `gorm:"column:storage_mode" json:"storage_mode"`
	StorageConfigID uint64 `gorm:"column:storage_config_id" json:"storage_config_id"`
	OriginName      string `gorm:"column:origin_name" json:"origin_name"`
	ObjectName      string `gorm:"column:object_name" json:"object_name"`
	Hash            string `gorm:"column:hash" json:"hash"`
	MimeType        string `gorm:"column:mime_type" json:"mime_type"`
	StoragePath     string `gorm:"column:storage_path" json:"storage_path"`
	Suffix          string `gorm:"column:suffix" json:"suffix"`
	SizeByte        int64  `gorm:"column:size_byte" json:"size_byte"`
	SizeInfo        string `gorm:"column:size_info" json:"size_info"`
	URL             string `gorm:"column:url" json:"url"`
	AuditColumns
	Timestamps
	Remark string `gorm:"column:remark" json:"remark"`
}

func (Attachment) TableName() string {
	return "attachment"
}

File: MineAdmin-web/src/modules/base/api/attachment.ts
/**
 * MineAdmin is committed to providing solutions for quickly building web applications
 * Please view the LICENSE file that was distributed with this source code,
 * For the full copyright and license information.
 * Thank you very much for using MineAdmin.
 *
 * @Author X.Mo<root@imoi.cn>
 * @Link   https://github.com/mineadmin
 */
import type { PageList, ResponseStruct } from '#/global'
import type { OperationParams, OperationRequest } from '@/generated/admin-api'

export interface AttachmentVo extends Omit<Partial<OperationParams<'adminAttachmentList'>>, 'suffix'> {
  id?: number
  storage_mode?: string
  storage_config_id?: number
  origin_name?: string
  object_name?: string
  hash?: string
  mime_type?: string
  storage_path?: string
  suffix?: string | string[]
  size_byte?: number
  size_info?: string
  url?: string
  remark?: string
}

export function upload(file: File): Promise<ResponseStruct<AttachmentVo>> {
  const formData = new FormData()
  const contractFile: OperationRequest<'adminAttachmentUpload'>['file'] = file
  formData.append('file', contractFile)
  return useHttp().post('/admin/attachment/upload', formData)
}

export function pageList(params: AttachmentVo): Promise<ResponseStruct<PageList<AttachmentVo>>> {
  return useHttp().get('/admin/attachment/list', { params })
}

export function deleteById(id: OperationParams<'adminAttachmentDelete'>['id']): Promise<ResponseStruct<null>> {
  return useHttp().delete(`/admin/attachment/${id}`)
}
